### ISSUE TITLE
Native client ignores filters

### ISSUE THREAD

**Issue by aquua** [clojurewerkz/elastisch#145]
sorry I can't write English.

watch my source.

``` clojure
(def rest-result (doc/search (core/conn*) "crawler" "product"
            :query (q/term :site_name "AAA")
            :filter {:range {:price {:from 0 :to 10000}}}
            :from 0
            :size 10))

(def native-result (native-doc/search core/tcp-conn "crawler" "product"
             :query (q/term :site_name "AAA")                         
             :filter {:range {:price {:from 0 :to 10000}}}
             :from 0
             :size 10))

(map #(get-in  % [:_source :price])(get-in rest-result [:hits :hits]))
-->  (6000 6000 6000 6000 6000 6000 6000 3000 6000 3000)

(map #(get-in  % [:_source :price])(get-in native-result [:hits :hits]))
-->(130000 100000 300000 6000 400000 6000 130000 310000 150000 50000)
```

REST API  Good work
Native API Not work

Test Version : 2.1.0 and 2.2.0-beta1

**Comment by aquua** [comment]
(def native-result (native-doc/search core/tcp-conn "crawler" "product"
                         :query (q/filtered :query (q/term :site_name "뽐뿌")
                                            :filter {:range {:price {:from 0 :to 10000}}}
                                            )
                         :from 0
                         :size 10))   

this source good work..

**Comment by michaelklishin** [comment]
Please be more specific than "it doesn't work".

**Comment by michaelklishin** [comment]
Is `:filter` ignored by the `elastisch.native.document/search`? Is there an exception? Please at least post the exception you're getting if so. I'm still not sure what the issue is.

**Comment by aquua** [comment]
I request price form  0  to  10000
rest api  price result    : 6000 6000 6000 6000 6000 6000 6000 3000 6000 3000 
native api price result  : 6000~ 150000

**Comment by michaelklishin** [comment]
OK, so the filter is ignored or not constructed correctly. Thanks for reporting.

**Comment by aquua** [comment]
thank you.. and sorry I can't write English..

**Comment by michaelklishin** [comment]
@aquua no worries. Your English is way better than my Korean ;)

**Comment by ronaldsuwandi** [comment]
The same with `missing` / `exists` filter. 

I have the following records
```
{:name "a" :email "a@a.com"}
{:name "b" :email nil}
{:name "c" :email nil}
```
and querying it directly from elastic search works fine `GET /uni_test/users/_search` returns all 3 records while 
```
GET /uni_test/users/_search
{
    "query": {
        "filtered": {
            "query": {
                "match_all": {}
            },
            "filter":{
                "missing": {
                    "field": "email"
                }
            }
        }        
    }
}
```
returns 2 records as expected, but when I do it in elastisch I keep getting all 3 records
```
(doc/search *es* test-index (name :users)
                          {:query {:filtered {:filter {:missing {:field :email}}
                                              :query {:match_all {}}}}})
```
I’ve tried with both `:filter` and `:query` as well as `:filtered` query, also tried using elastisch 2.1.0, 2.2.0-beta4 and even upgraded elasticsearch client to 1.7.2 without any luck

This only fails on the native driver, rest driver works fine

**Comment by ronaldsuwandi** [comment]
any news on this bug?

**Comment by michaelklishin** [comment]
@ronaldsuwandi I hope to fix it before 2.2.0. This is open source software, however, so if you need it fixed soon, please consider investing your own time.

**Comment by michaelklishin** [comment]
Looks like it may be a matter of aliasing `:post-filter` to `:filter` in the native client. Java client has renamed `.setFilter` to `.setPostFilter` at some point as far as I can tell. Will continue digging tomorrow.

**Comment by ronaldsuwandi** [comment]
thanks @michaelklishin . I'll take a look as well

**Comment by michaelklishin** [comment]
OK, that was indeed it: option name was off and one test that performed filtering had an incorrect expectation since April 2014 (!).

This is really embarrassing but oh well. Fixed in master, I'm going to release `2.2.0-rc1` over the w/e.
